stdlib/public/SwiftShims/CMakeLists.txt

```diff
@@ -106,34 +84,47 @@ function(find_llvm_clang_headers suffix description out_var)
 endfunction()
 
 if(SWIFT_BUILT_STANDALONE)
-  find_llvm_clang_headers("/clang/${CLANG_VERSION}" "Clang"
-    clang_headers_location)
+  find_llvm_clang_headers("/clang/${CLANG_VERSION}" "Clang" clang_headers_location)
 else() # NOT SWIFT_BUILT_STANDALONE
   set(clang_headers_location "${LLVM_LIBRARY_DIR}/clang/${CLANG_VERSION}")
 endif()
 
 if("${CMAKE_SYSTEM_NAME}" STREQUAL "Windows")
-  set(cmake_symlink_option "copy_directory")
+  set(cmake_symlink_option copy_directory)
 else()
-  set(cmake_symlink_option "create_symlink")
+  set(cmake_symlink_option create_symlink)
 endif()
 
 add_custom_command_target(unused_var
-    COMMAND
-      "${CMAKE_COMMAND}" "-E" "make_directory" "${SWIFTLIB_DIR}"
-    COMMAND
-      "${CMAKE_COMMAND}" "-E" "${cmake_symlink_option}"
-      "${clang_headers_location}"
-      "${SWIFTLIB_DIR}/clang"
-
-    CUSTOM_TARGET_NAME "symlink_clang_headers"
-    OUTPUT "${SWIFTLIB_DIR}/clang"
-    COMMENT "Symlinking Clang resource headers into ${SWIFTLIB_DIR}/clang")
-add_dependencies(copy_shim_headers symlink_clang_headers)
+                          OUTPUT
+                            ${SWIFTLIB_DIR}/clang
+                          CUSTOM_TARGET_NAME
+                            swift-clang-resource-dir-symlink
+                          COMMAND
+                            ${CMAKE_COMMAND} -E make_directory ${SWIFTLIB_DIR}
+                          COMMAND
+                            ${CMAKE_COMMAND} -E ${cmake_symlink_option} ${clang_headers_location} ${SWIFTLIB_DIR}/clang)
+
+add_custom_command_target(unused_var
+                          OUTPUT
+                            ${SWIFTLIB_DIR}/shims
+                          CUSTOM_TARGET_NAME
+                            swift-shim-headers
+                          COMMAND
+                            ${CMAKE_COMMAND} -E make_directory ${SWIFTLIB_DIR}/shims
+                          COMMAND
+                            ${CMAKE_COMMAND} -E copy_if_different ${sources} ${SWIFTLIB_DIR}/shims/
+                          WORKING_DIRECTORY
+                            ${CMAKE_CURRENT_SOURCE_DIR}
+                          DEPENDS
+                            ${sources})
 
 swift_install_in_component(compiler
-    FILES ${sources}
-    DESTINATION "lib/swift/shims")
+                           FILES ${sources}
+                           DESTINATION "lib/swift/shims"
+                           COMPONENT swift-shim-headers)
+add_llvm_install_targets(install-swift-shim-headers
```

Comment:
Can you add a comment here that this is needed so we can eventually move from swift components to llvm install components and that today this makes install-distribution work in the unified build?